src/core/qgsabstractdatabaseproviderconnection.cpp

```diff
@@ -0,0 +1,280 @@
+/***************************************************************************
+  qgsabstractdatabaseproviderconnection.cpp - QgsAbstractDatabaseProviderConnection
+
+ ---------------------
+ begin                : 2.8.2019
+ copyright            : (C) 2019 by Alessandro Pasotti
+ email                : elpaso at itopen dot it
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "qgsabstractdatabaseproviderconnection.h"
+#include "qgsexception.h"
+#include <QVariant>
+#include <QObject>
+
+QgsAbstractDatabaseProviderConnection::QgsAbstractDatabaseProviderConnection( const QString &name ):
+  QgsAbstractProviderConnection( name )
```

Comment:
Just a minor nitpick - QGIS coding convention is:

   Class::Class()
      : mMember( value )
      , mMember2( value )
...
etc